### DISCUSSION TITLE
Suggestion on defining domains with different magnitude

### DISCUSSION THREAD

**Discussion by WJiangH** [lululxvi/deepxde#1578] [Q&A]
Hi,

I am dealing with solving PDE systems (a forward problem) where the time domain is from 0 to 1e5 (s) and the x domain is from 0 to 100 nm. The magnitude of two domains is quite different. Any suggestions on selecting training data, for example, dt and dx? Or normalization on both domain needed?

Thanks!!

Best,
JJ

**Comment by ryabs07** [comment]
Hi, 
From my experience with straight forward PINN, you have to normalize PDEs with respect to both space and time for these kind of problems. But even then, for the case of time scale ~ 1e5 seconds, the solution is not converging for me. If you find any other alternative appproach, please do share.
Best of luck.

**Reply by WJiangH** [reply]
Thank you for sharing. 

Yes, I believe non-dimensionalization would be necessary for this case for my domains. 

Here is also another concern in my case, in which the target output have quite different magnitudes, for example, initial or  boundary conditions: u1=1e-40, u2=1e-14, u3=0.2, u4=0.8. The gradients of u1 and u2 have significant impact to u3 and u4. It causes a problem in defining the loss functions, where some terms in final loss have very large value while some are very small. 

I am considering using the log to scale the output. However, to make the underlying PDEs alway valid, the log operation may also need be applied to the original PDEs. 

Appreciate it if you can share any ideas for the problem mentioned above.

Best,
JJ

**Reply by lululxvi** [reply]
Yes, the idea is basically change of variable.
